website/server/libs/spells.js

```diff
@@ -163,14 +163,32 @@ async function castSpell (req, res, { isV3 = false }) {
       task: results[1],
     });
   } else if (targetType === 'self') {
-    await castSelfSpell(req, user, spell, quantity);
-
-    let userToJson = user;
-    if (isV3) userToJson = await userToJson.toJSONWithInbox();
-
-    res.respond(200, {
-      user: userToJson,
-    });
+    // Check if chilling frost or stealh skill has been previously casted or not.
+    // See #12361 for more details.
+    const spellName = spell.key;
+    const incompleteDailiesDue = await Tasks.Task.find({
+      userId: user._id,
+      type: 'daily',
```

Comment:
can the query be simplified into: 

```
{
    userId: user._id,
    type: 'daily',
    completed: false,
    isDue: true,
}
```

since by default mongo treats all properties of the query as ANDs

Reply:
Modified the query.